### PR TITLE
IBX-665: Bumped Symfony dependency to 5.3

### PR DESCRIPTION
**Pull request by webhdx**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [IBX-665](https://issues.ibexa.co/browse/IBX-665)
| **Improvement**| yes
| **New feature**    | no
| **Target version** | `2.3`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: src/EzPlatformCoreBundle/bundle/DependencyInjection/Compiler/SessionConfigurationPass.php
Comment (Steveb-p) [inline src/EzPlatformCoreBundle/bundle/DependencyInjection/Compiler/SessionConfigurationPass.php]: We should check for each & every service separately, and replace argument in each instance.

This way we avoid the assumption that when `session.storage.native` exists, there will also be a `session.storage.php_bridge`. And if somehow all 4 exist, then it will apply to all 4, not just 2.

Comment (webhdx) [reply src/EzPlatformCoreBundle/bundle/DependencyInjection/Compiler/SessionConfigurationPass.php]: It's just a copy-paste from Symfony FrameworkExtension but I see your point. I will change it soon.

Comment (webhdx) [reply src/EzPlatformCoreBundle/bundle/DependencyInjection/Compiler/SessionConfigurationPass.php]: Fixed: db17482

**Review by Steveb-p — APPROVED** [review]
You might want to make the test more abstract now (testing compiler pass each service separately, with all 4 and maybe with none). :+1: nonetheless.

**Review by adamwojs — APPROVED** [review]

**Comment by Steveb-p** [comment]
Merging this will unblock tests that use the `IbexaTestKernel`.

**Comment by Steveb-p** [comment]
Currently blocks ibexa/migration `master` branch tests and the new commerce. Shall we merge, or are we waiting for more reviews / QA?

**Comment by mnocon** [comment]
@Steveb-p there's a strange issue on Travis where the browser tests are not able to log in on some setups on Symfony 5.3, something @webhdx will need to investigate (next week I guess?)

**Comment by lserwatka** [comment]
Yes, we are not ready for SF 5.3. We need to debug issue @mnocon mentioned.

**Comment by Steveb-p** [comment]
Could we split the version bump and the changes to the compiler pass?

As it stands it blocks the integration tests on anything that uses application kernel for that (due to session handlers not being defined).

I wanted to fix that with #41, but since this PR covers the same code (and expands upon changes, with 5.3 compatibility) I've closed that one.

**Comment by Steveb-p** [comment]
Rebased after #41.

**Review by mnocon — APPROVED** [review]

**Pull request merged**
